apps/passport-client/src/defaultSubscriptions.ts

```diff
@@ -61,24 +82,36 @@ export async function addDefaultSubscriptions(
     );
   }
 
-  for (const feedUrl of DEFAULT_FEED_URLS) {
-    const { feeds, providerName, providerUrl } =
-      await subscriptions.listFeeds(feedUrl);
+  const subscriptionPromises = [];
 
-    if (!subscriptions.hasProvider(feedUrl)) {
-      subscriptions.addProvider(providerUrl, providerName);
+  for (const feedUrl of DEFAULT_FEED_URLS) {
+    // DEFAULT_FEED_URLS is a list of URLs to individual feeds
+    // The reason we have to use URL strings is because Zupass doesn't know
+    // about the feed objects in the way that it does for feeds that come from
+    // the Zupass backend (see zupassDefaultSubscriptions above).
+    //
+    // In principle, multiple feeds can live at the same URL. In practice, we
+    // do not do this for feeds served by Podbox (which is where all of the
+    // feeds in DEFAULT_FEED_URLS come from). So, the final part of the URL is
+    // always the feed ID. Given this, we can check to see if the user is
+    // already subscribed to this feed: if so, we do not need to re-subscribe
+    // them.
+    const feedId = feedUrl.split("/").pop();
```

Comment:
i worry that this shortcut could come to bite us later if we add a feedUrl in DEFAULT_FEED_URLS that has multiple feeds or doesn't fit this pattern exactly, e.g. if we ever change Podbox feed issuance or subscribe to non-Podbox feeds

Reply:
i don't have a much better solution off the top of my head, but perhaps more motivating to just rethink the DEFAULT_FEED_URLS approach to begin with, e.g.

https://linear.app/0xparc-pcd/issue/0XP-752/make-starred-pipelines-on-podbox-be-auto-subscribed-on-zupass-client